UPGRADE.md

```diff
@@ -34,6 +39,31 @@ As part of the update of flysystem the support for the guzzle client package `gu
 
 The `GoogleGeolocator` and the `NominatimGeolocator` no longer support the Guzzle client and require a `Symfony\HttpClient` client instead.
 
+### Updating to flysystem 3
+
+The Sulu media storage uses now the Flysystem Bundle which need to be configured:
+
+```yaml
+# config/packages/flysystem.yaml
+flysystem:
+    storages:
+        default.storage:
+            adapter: 'local'
+            options:
+                directory: '%kernel.project_dir%/var/uploads'
+```
+
+If you want use a [different storage](https://github.com/thephpleague/flysystem-bundle/blob/3.x/docs/2-cloud-storage-providers.md) for Sulu you can configure it here:
+
+```yaml
+# config/packages/sulu_media.yaml
+sulu_media:
+    storage:
+        flysystem_service: 'default.storage' # this is default and not required to be configured.
+```
+
+This will only create the service `sulu_media.storage` as the alias to `sulu_media.storage.*` services has been removed.
+
```

Comment:
I would like to have maybe also example in the upgrade.md how to migrate the existing s3, google cloud storage and azure blob storage examples to Flysystem config:

### Sulu Media S3 to Flysystem S3

**Before*:

```
sulu_media:
    storage: s3
    storages:
        s3:
            key: 'your aws s3 key'
            secret: 'your aws s3 secret'
            bucket_name: 'your aws s3 bucket name'
            path_prefix: 'optional path prefix'
            region: 'eu-west-1'
```

**New**:

```
flysystem:
    storages:
        default.storage:
             # ...
```


### Sulu Google Cloud Config to Flysystem S3

**Before*:

```
sulu_media:
    storage: google_cloud
    storages:
        google_cloud:
            key_file_path: '/path/to/key.json'
            bucket_name: 'sulu-bucket'
            path_prefix: 'optional path prefix'
```

**New**:

```
flysystem:
    storages:
        default.storage:
             # ...
```



### Sulu Azure Config to Flysystem Config

**Before*:

```
sulu_media:
    storage: google_cloud
    storages:
        google_cloud:
            key_file_path: '/path/to/key.json'
            bucket_name: 'sulu-bucket'
            path_prefix: 'optional path prefix'
```

**New**:

```
flysystem:
    storages:
        default.storage:
             # ...
```


https://docs.sulu.io/en/2.6/cookbook/external-media-storage.html